### DISCUSSION TITLE
MediaPortal2 Plugin Discussion

### DISCUSSION THREAD

**Discussion by da3dsoul** [ShokoAnime/ShokoServer#903] [Ideas]
@RickDB sorry for the ping, I have no idea how to contact you. I noticed that you have been a plugin dev for MP2, though I heard you switched to Plex/Kodi through whatever wizardry is involved in that plugin. I know you haven't worked on Shoko even in the time that it's been called Shoko, but hopefully you're still lurking as a happy, or at least tolerant, user somewhere.

Kodi's plugin API has pissed me off for a long time, and they don't have the dev time to care about improving it. I've been considering switching to something else, as, due to updates and obsolescence, eventually I'll either need to make a new Kodi plugin or switch.

What's your experience working with them? Why did you switch to Plex? What issues do you think I'd have if I tried making an MP2 plugin for Shoko? Ideally, I'd want it to stream like the MP1 plugin did, rather than try to hook into the internal library like our Plex plugin does. I wouldn't try to go for a full client like the MP1 plugin did, just a light client with maybe some quick actions.

Obviously, anyone else with info can pop in.

**Comment by RickDB** [comment]
MP2 in the long run is a dead end in my opinion with the presence of Plex and Emby / Jellyfin which is one of the reason I didn't put a lot of hours into it back then.
The MP2 core design is solid though and developers are friendly and capable so writing plugins for it will be a breeze compared to Kodi's absolutely dreadful API, you will however never get any real user base for it to make it feel worth it in my opinion :)

MP1 was a different matter some time ago as it had more developers but progress slowed down and few core plugin developers that kept it going have now moved on as well, you will run into the same thing as with MP2 and that is very small user base.

With both MP1 and MP2 it's a hard decision as you need to start somewhere and comes down to is it worth the time and effort.
Compared to Kodi maintaining a MP1/2 plugin requires far less maintenance (skin updates mostly) and won't get annoyed by the API so that is something to consider as well.

For the Plex part:

Dropped Kodi + PlexKodiConnect some time ago in favor of Plex apps (Apple TV / Plex Media Player) as they matured to a point they were ideal for my use case.
Also with Kodi madVR integration no longer being worked on (last release for 17.x) and Kodi lacking proper HDR and upscaling support no reason to keep it around.

My other reasons for switching to Plex:

- Ease of management and Docker / Linux server support (run it on Archlinux + ZFS for storage)
- Great app support on the Apple platform and can basically run on any device 
- Share libraries with friends and watch together function is awesome especially in these times
- Stream from everywhere
- Third party libraries like PlexAPI for ease of development
- Plenty of apps like Tautulli / Varys for statistics and overview
- Anime agent support thru HAMA with near perfect matching
- Less time for personal projects due to new job function wanted something less maintenance and "just work" 

Developed PlexAniSync to do the list syncing to AniList as a stop gap which while not perfect for all series gets the job done, needs some TLC still with matching based on HAMA's but motivation and time is low ;)

When developing anything Plex would go with external integration because at some point they will fully drop plugin support and maybe even dial back on third party agent support.
They don't have the best track record when it comes to long term support or clear roadmaps.

Anyway just my 2 cents and happy to pitch in :)

**Comment by da3dsoul** [comment]
Thank you much. I'll be honest. I don't care how many people use it. I make clients for me. I make Server is for everyone. If they have dedicated and friendly devs, that's all I really care about.
I know that we still have users that have wanted MP support, to the point that they don't update Shoko to keep it working. For every one person that's vocal about it, there's a dozen that lurk.

**Comment by da3dsoul** [comment]
My dislike of Plex is a personal one. Jellyfin I'll need to look at and see if I can customize the appearance to my liking.

**Comment by Andy2244** [comment]
Hey, i also just revisited my media setup and was looking for a solution with working MadVR, Trakt + AniDB watched status syncing. The main problem is that using the 17.7 [Kodi-DSplayer](https://forum.kodi.tv/showthread.php?tid=351534) is already problematic, since i already noticed issues with the 17.x trakt plugin.

So ideally i need a setup with working MadVR, but also on a somewhat maintained/updated branch.

1) How does the "Plex Media Player" handle MadVR, just via simple external player like Kodi? I tried DSplayer again, so watch detection works and resume too. The way Kodi handles external players is just bad, yet DSplayer is no longer developed.

2) Can then Plex trakt plugin sync your watched status all in one go, similar to what the Kodi plugin does? So a fresh install can sync everything, without manual backup/restores?

Regarding MP, i just installed MP1 and oh boy i forgot how ancient and crappy the setup/config process is. I had a big laughy at the "Now go to this link and download Windows Media Player 11 x86 installer and put it in the temp install dir" step...

PS: I guess i need to give Plex a testrun, since using MP1 seems also like a dead-end.

**Comment by RickDB** [comment]
Hi,

**1. How does the "Plex Media Player" handle MadVR, just via simple external player like Kodi? I tried DSplayer again, so watch detection works and resume too. The way Kodi handles external players is just bad, yet DSplayer is no longer developed.**

No madVR support at all but you can tweak its MPV settings to provide very decent down / upscaling results including some popular machine learning profiles.

**2. Can then Plex trakt plugin sync your watched status all in one go, similar to what the Kodi plugin does? So a fresh install can sync everything, without manual backup/restores?**

Hit an miss for my Anime library when it comes to syncing to Trakt.
Expect main reason is HAMA agent with different metadata not being supported but never fully investigated this to be honest.

Jellyfin is a good alternative and since MrMC got an update with support for gonna give it another swing.
My dislike for Plex is also there with their poor UI decisions and development time spend in the "wrong" areas, keep getting the feeling that they are:

- Short on financial resources, hence pushing their third part integrations like Tidal / WB aggresively
- Preparing to sell their company in the near future if above fails

Right now Plex has the most complete package and ease of use makes it a very attractive option just unclear how long that will last.

**Comment by Andy2244** [comment]
> provide very decent down / upscaling

Problem is i mainly need working "Smooth Motion", since my old plasma handles 23/24p badly. So Anime panning looks horrible without MadVR's "magic" on 59/60hz.

> Right now Plex has the most complete package and ease of use makes it a very attractive option just unclear how long that will last.

Mhh this would make MP2 more attractive or even Jellyfin. Need to research how the Anime support is for Jellyfin...

**Comment by da3dsoul** [comment]
Rick, if you want to rejoin the team, we could use someone that actually cares about Trakt. Our systems blow every other option, especially HAMA, away. Trakt works, I think? It needs some love, though.

**Comment by da3dsoul** [comment]
@Andy2244 we have a WIP plugin for Jellyfin, and we have a member of their team in the discord.

**Comment by Cazzar** [comment]
> Problem is i mainly need working "Smooth Motion", since my old plasma handles 23/24p badly. So Anime panning looks horrible without MadVR's "magic" on 59/60hz.

I would see that causing issues with more than just anime... and honestly poor design on the general status of the TV in full, since a lot of movies are in 24fps commonly.

Though with plex, it can depend on your playback client there are third party plex clients that work, or in some case people just do playback by things like chromecasts and similar.

Though ultimately with this discussion, I think we look at providing a rich API that others can make their own clients on top, even if it requires a plugin, as some of the plugin API work that I have done even allows custom API controllers within Shoko.

**Comment by RickDB** [comment]
> 
> 
> Rick, if you want to rejoin the team, we could use someone that actually cares about Trakt. Our systems blow every other option, especially HAMA, away. Trakt works, I think? It needs some love, though.

Workload right now is hefty with a new security department and educating colleagues to offload some of my tasks, late September / early October it should settle down and would love working on the Trakt area then though :)

**Comment by da3dsoul** [comment]
Cool. Obviously, life comes first. Our TvDB system is awesome, so you can use that as a base for linking

**Comment by Andy2244** [comment]
> we have a WIP plugin for Jellyfin

@da3dsoul You have a github repo for it or a testversion? I can't really find much about Jellyfin and Shoko?

**Comment by da3dsoul** [comment]
One of our power users was working on one. AFAIK it's not in a working condition yet. Ping wispers on the discord (in general or support) for info.

**Comment by da3dsoul** [comment]
Can confirm Shoko MP2 plugin is happening. Also, I'm sort of an MP2 dev now. The userbase may not be huge, but it's not like any of us have the time for millions of users like Plex or Kodi has.
@jmediamanager figure you use Plex like just about everyone else, but I thought you might be interested in knowing.

**Comment by speedello** [comment]
@da3dsoul Ah that's cool. Yes I am using Plex now, but still have a soft spot for MP :)
